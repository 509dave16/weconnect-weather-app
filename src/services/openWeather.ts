
import _ from 'lodash'

import { ApiResponse } from '../interfaces/ApiResponse'
import { ApiResponseState, Period } from '../interfaces/Enums'
import { CurrentWeather } from '../interfaces/CurrentWeather'
import { DailyForecast } from '../interfaces/DailyForecast'

import { OPEN_WEATHER_API_URL, OPEN_WEATHER_APP_ID } from '../config/env'

function getDateString(forecast: any): string {
  return forecast?.dt_txt?.split('T')?.[0]
}

function getCurrentWeatherFromJson(json: any): CurrentWeather {
  return {
    location: json?.name,
    temperature: json?.main?.temp,
    feelsLike: json?.main?.feels_like,
    wind: json?.wind,
    condition: json?.weather?.[0]?.main,
  }
}

function getDailyForecast(name: String, dailyForecastGroup: Array<any>): DailyForecast {
  // console.log(name, dailyForecastGroup)
  const amForecast: any = _.minBy(dailyForecastGroup, (forecast) => forecast.main.temp)
  const pmForecast: any = _.maxBy(dailyForecastGroup, (forecast) => forecast.main.temp) 
  // console.log('<<<AM FORECAST', amForecast)
  // console.log('<<<PM FORECAST', pmForecast)
  const amCurrentWeather: CurrentWeather = getCurrentWeatherFromJson({ ...amForecast, name })
  const pmCurrentWeather: CurrentWeather = getCurrentWeatherFromJson({ ...pmForecast, name })
  return {
    date: new Date(getDateString(amForecast)),
    periods: [
      { period: Period.AM, ...amCurrentWeather },
      { period: Period.PM, ...pmCurrentWeather },
    ]
  }
}

function getDailyForecastsFromJson(json: any): Array<DailyForecast> {
  // console.log('<<<getDailyForecastsFromJson -', json)
  const { city: { name, timezone }, sunrise, sunset, list } = json
  const sunRisingAt = new Date(sunrise * 1000)
  const sunSettingAt = new Date(sunset * 1000)
  // console.log('<<<sunRisingAt/sunSettingAt', sunRisingAt, sunSettingAt)
  list.forEach((forecast: any) => { forecast.dt_txt = (new Date((forecast.dt + timezone) * 1000)).toISOString() })
  const dailyForecastGroups = _.groupBy(list, getDateString)
  return _.values(dailyForecastGroups)
    .filter((forecastGroup: Array<any>) => forecastGroup.length > 1)
    .map((forecastGroup: Array<any>) => getDailyForecast(name, forecastGroup))
}

function getRequest(resource: String, options: RequestInit): Request {
  let url = `${OPEN_WEATHER_API_URL}/${resource}`
  if (!options.method || ['HEAD', 'GET'].includes(options.method) && options.body instanceof URLSearchParams) {
    url += `?${options.body?.toString()}`
    delete options.body
  }
  return new Request(url, options)
}

async function openWeatherRequest(resource: String, options: RequestInit = {}, transformData: Function): Promise<ApiResponse> {
  let state: ApiResponseState = ApiResponseState.SUCCESS
  let result: any = null

  // console.log('<<<openWeatherRequest - options.body', options?.body)
  if (options?.body instanceof URLSearchParams) {
    options.body.append('appid', OPEN_WEATHER_APP_ID)
  }

  try {
    const response: Response = await fetch(getRequest(resource, options));
    const json: JSON = await response.json();

    if (response.status >= 400) {
      state = ApiResponseState.ERROR
      result = json
    } else if (_.isEmpty(json)) {
      state = ApiResponseState.EMPTY
      result = null
    } else {
      result = transformData(json)
    }
  } catch (error) {
    state = ApiResponseState.ERROR
    result = error
  }

  return { result, state }
}

export async function currentWeather(options: RequestInit): Promise<ApiResponse> {
 return openWeatherRequest('weather', options, getCurrentWeatherFromJson)
}

export function dailyForecasts(options: RequestInit): Promise<ApiResponse> {
  return openWeatherRequest('forecast', options, getDailyForecastsFromJson)
}