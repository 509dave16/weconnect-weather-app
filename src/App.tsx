import React from 'react';
import { StyleSheet, Text, ScrollView, View, ActivityIndicator, Dimensions } from 'react-native';
import { StatusBar } from 'expo-status-bar';
import { SafeAreaView } from 'react-native-safe-area-context'
import Weather from './components/Weather';
import Forecasts from './components/Forecasts';

import * as OpenWeather from './services/openWeather';
import { ApiResponse } from './interfaces/ApiResponse';
import { ApiResponseState } from './interfaces/Enums';
import { CurrentWeather } from './interfaces/CurrentWeather';
import { DailyForecast } from './interfaces/DailyForecast';

const location = 'Spokane'
const backgroundColor = '#b9e1f8'

interface State {
  currentWeatherDone: boolean;
  dailyForecastsDone: boolean;
  currentWeather?: CurrentWeather;
  dailyForecasts?: Array<DailyForecast>;
}

export default class App extends React.Component<{}, State> {
  constructor(props: any) {
    super(props)
    this.state = { currentWeatherDone: false, dailyForecastsDone: false }
  }

  componentDidMount() {
    this.fetchCurrentWeather();
    this.fetchDailyForecasts();
  }

  async fetchCurrentWeather() {
    const response: ApiResponse = await OpenWeather.currentWeather({
      body: new URLSearchParams({
        q: location,
        units: 'imperial'
      })
    })

    const nextState: any = { currentWeatherDone: true }
    if (response.state === ApiResponseState.SUCCESS) {
      nextState.currentWeather = response.result
    }
    this.setState(nextState)
    console.log('<<<fetchCurrentWeather - response', response)
  }

  async fetchDailyForecasts() {
    const response: ApiResponse = await OpenWeather.dailyForecasts({
      body: new URLSearchParams({
        q: location,
        units: 'imperial'
      })
    })
    const nextState: any = { dailyForecastsDone: true }
    if (response.state === ApiResponseState.SUCCESS) {
      nextState.dailyForecasts = response.result
    }
    this.setState(nextState)
    // console.log('<<<fetchDailyForecasts - response', response)
  }

  shouldRenderContent() {
    return this.state.currentWeatherDone && this.state.dailyForecastsDone
  }

  render() {
    return (
      <SafeAreaView style={styles.container}>
        { this.shouldRenderContent() ? this.renderContent() : this.renderStatus() }
        <StatusBar backgroundColor={backgroundColor} style='dark' />
      </SafeAreaView>
    )
  }

  renderContent() {
    return (
      <ScrollView contentContainerStyle={styles.contentContainer}>
        <View style={styles.contentSpacing}></View>
        <Weather currentWeather={this.state.currentWeather} />
        <Forecasts dailyForecasts={this.state.dailyForecasts} />
      </ScrollView>
    )
  }

  renderStatus() {
    return (
      <View style={styles.statusContainer}>
        <ActivityIndicator size='large' />
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor,
  },
  contentSpacing: {
    marginTop: Dimensions.get('screen').height / 8,
  },
  contentContainer: {
    flexGrow: 1,
    paddingHorizontal: 25,
    alignItems: 'center',
  },
  statusContainer: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
});
