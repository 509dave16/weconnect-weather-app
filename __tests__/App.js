import 'react-native'
import React from 'react'
import App from '../src/App.tsx'

// Note: test renderer must be required after react-native.
import { render, waitForElementToBeRemoved } from '@testing-library/react-native'

import * as WeatherService from '../src/services/weather'
jest.mock('../src/services/weather')

beforeEach(() => {
  const currentWeatherResponse = {
    "result": {
      "condition": "Clear",
      "feelsLike": 55.85,
      "location": "Spokane",
      "temperature": 60.01,
      "wind": {
        "deg": 50,
        "speed": 3.36,
      },
    },
    "state": "success",
  }

  const dailyForecastsResponse = {
    result: [
      {
        "date": new Date("2020-09-11T00:00:00.000Z"),
        "periods": [
          {
            "condition": "Clear",
            "feelsLike": 53.55,
            "location": "Spokane",
            "period": "AM",
            "temperature": 58.33,
            "wind": {
              "deg": 111,
              "speed": 2.55,
            },
          },
          {
            "condition": "Clear",
            "feelsLike": 78.3,
            "location": "Spokane",
            "period": "PM",
            "temperature": 85.32,
            "wind": {
              "deg": 240,
              "speed": 6.24,
            },
          },
        ],
      },
      {
        "date": new Date("2020-09-12T00:00:00.000Z"),
        "periods": [
          {
            "condition": "Clear",
            "feelsLike": 53.82,
            "location": "Spokane",
            "period": "AM",
            "temperature": 59.61,
            "wind": {
              "deg": 116,
              "speed": 3.74,
            },
          },
          {
            "condition": "Clear",
            "feelsLike": 77.88,
            "location": "Spokane",
            "period": "PM",
            "temperature": 85.1,
            "wind": {
              "deg": 214,
              "speed": 6.98,
            },
          },
        ],
      },
      {
        "date": new Date("2020-09-13T00:00:00.000Z"),
        "periods": [
          {
            "condition": "Clouds",
            "feelsLike": 51.96,
            "location": "Spokane",
            "period": "AM",
            "temperature": 58.15,
            "wind": {
              "deg": 92,
              "speed": 4.32,
            },
          },
          {
            "condition": "Clouds",
            "feelsLike": 80.82,
            "location": "Spokane",
            "period": "PM",
            "temperature": 88.18,
            "wind": {
              "deg": 246,
              "speed": 6.02,
            },
          },
        ],
      },
      {
        "date": new Date("2020-09-14T00:00:00.000Z"),
        "periods": [
          {
            "condition": "Clouds",
            "feelsLike": 53.71,
            "location": "Spokane",
            "period": "AM",
            "temperature": 62.1,
            "wind": {
              "deg": 195,
              "speed": 11.12,
            },
          },
          {
            "condition": "Clouds",
            "feelsLike": 65.77,
            "location": "Spokane",
            "period": "PM",
            "temperature": 75.31,
            "wind": {
              "deg": 225,
              "speed": 12.01,
            },
          },
        ],
      },
      {
        "date": new Date("2020-09-15T00:00:00.000Z"),
        "periods": [
          {
            "condition": "Clouds",
            "feelsLike": 57.13,
            "location": "Spokane",
            "period": "AM",
            "temperature": 62.46,
            "wind": {
              "deg": 139,
              "speed": 4.18,
            },
          },
          {
            "condition": "Clouds",
            "feelsLike": 70.56,
            "location": "Spokane",
            "period": "PM",
            "temperature": 73.17,
            "wind": {
              "deg": 244,
              "speed": 2.39,
            },
          },
        ],
      },
    ],
    "state": "success",
  }
  WeatherService.currentWeather.mockResolvedValue(currentWeatherResponse)
  WeatherService.dailyForecasts.mockResolvedValue(dailyForecastsResponse)
})

it('renders loading state correctly', () => {
  let root = render(<App />)
  expect(root.toJSON()).toMatchSnapshot()
})

it('renders data correctly', async () => {
  const { getByTestId, toJSON } = render(<App />)
  await waitForElementToBeRemoved(() => getByTestId('App-ActivityIndicator'))
  expect(toJSON()).toMatchSnapshot()
})