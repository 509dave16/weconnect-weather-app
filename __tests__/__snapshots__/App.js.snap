// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders data correctly 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "backgroundColor": "#b9e1f8",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "flexGrow": 1,
        "paddingHorizontal": 25,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "marginTop": 166.75,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 40,
            }
          }
        >
          Spokane
        </Text>
        <Text
          style={
            Object {
              "fontSize": 18,
            }
          }
        >
          Clear
        </Text>
        <Text
          style={
            Object {
              "fontSize": 60,
            }
          }
        >
          60
          °
        </Text>
        <View
          style={
            Object {
              "marginTop": 83.375,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 0,
                  "width": "50%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "textAlign": "left",
                  }
                }
              >
                Feels Like
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 0,
                  "width": "50%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "textAlign": "left",
                  }
                }
              >
                56
                °
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 0,
                  "width": "50%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "textAlign": "left",
                  }
                }
              >
                Wind
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 0,
                  "width": "50%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "textAlign": "left",
                  }
                }
              >
                50
                ° / 
                3.36
                 mph
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={Object {}}
      >
        <Text
          style={
            Object {
              "fontSize": 18,
              "marginVertical": 41.6875,
            }
          }
        >
          Forecast:
        </Text>
        <View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "marginBottom": 5,
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 0,
                  "width": "40%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "textAlign": "left",
                  }
                }
              >
                Thursday
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 0,
                  "width": "20%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "textAlign": "left",
                  }
                }
              >
                AM
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 0,
                  "width": "20%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "textAlign": "left",
                  }
                }
              >
                Clear
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 0,
                  "width": "20%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "textAlign": "left",
                  }
                }
              >
                58
                °
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "marginBottom": 5,
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 0,
                  "width": "40%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "textAlign": "left",
                  }
                }
              >
                Thursday
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 0,
                  "width": "20%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "textAlign": "left",
                  }
                }
              >
                PM
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 0,
                  "width": "20%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "textAlign": "left",
                  }
                }
              >
                Clear
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 0,
                  "width": "20%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "textAlign": "left",
                  }
                }
              >
                85
                °
              </Text>
            </View>
          </View>
        </View>
        <View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "marginBottom": 5,
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 0,
                  "width": "40%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "textAlign": "left",
                  }
                }
              >
                Friday
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 0,
                  "width": "20%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "textAlign": "left",
                  }
                }
              >
                AM
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 0,
                  "width": "20%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "textAlign": "left",
                  }
                }
              >
                Clear
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 0,
                  "width": "20%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "textAlign": "left",
                  }
                }
              >
                60
                °
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "marginBottom": 5,
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 0,
                  "width": "40%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "textAlign": "left",
                  }
                }
              >
                Friday
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 0,
                  "width": "20%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "textAlign": "left",
                  }
                }
              >
                PM
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 0,
                  "width": "20%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "textAlign": "left",
                  }
                }
              >
                Clear
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 0,
                  "width": "20%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "textAlign": "left",
                  }
                }
              >
                85
                °
              </Text>
            </View>
          </View>
        </View>
        <View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "marginBottom": 5,
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 0,
                  "width": "40%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "textAlign": "left",
                  }
                }
              >
                Saturday
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 0,
                  "width": "20%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "textAlign": "left",
                  }
                }
              >
                AM
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 0,
                  "width": "20%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "textAlign": "left",
                  }
                }
              >
                Clouds
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 0,
                  "width": "20%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "textAlign": "left",
                  }
                }
              >
                58
                °
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "marginBottom": 5,
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 0,
                  "width": "40%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "textAlign": "left",
                  }
                }
              >
                Saturday
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 0,
                  "width": "20%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "textAlign": "left",
                  }
                }
              >
                PM
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 0,
                  "width": "20%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "textAlign": "left",
                  }
                }
              >
                Clouds
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 0,
                  "width": "20%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "textAlign": "left",
                  }
                }
              >
                88
                °
              </Text>
            </View>
          </View>
        </View>
        <View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "marginBottom": 5,
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 0,
                  "width": "40%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "textAlign": "left",
                  }
                }
              >
                Sunday
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 0,
                  "width": "20%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "textAlign": "left",
                  }
                }
              >
                AM
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 0,
                  "width": "20%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "textAlign": "left",
                  }
                }
              >
                Clouds
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 0,
                  "width": "20%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "textAlign": "left",
                  }
                }
              >
                62
                °
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "marginBottom": 5,
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 0,
                  "width": "40%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "textAlign": "left",
                  }
                }
              >
                Sunday
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 0,
                  "width": "20%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "textAlign": "left",
                  }
                }
              >
                PM
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 0,
                  "width": "20%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "textAlign": "left",
                  }
                }
              >
                Clouds
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 0,
                  "width": "20%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "textAlign": "left",
                  }
                }
              >
                75
                °
              </Text>
            </View>
          </View>
        </View>
        <View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "marginBottom": 5,
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 0,
                  "width": "40%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "textAlign": "left",
                  }
                }
              >
                Monday
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 0,
                  "width": "20%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "textAlign": "left",
                  }
                }
              >
                AM
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 0,
                  "width": "20%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "textAlign": "left",
                  }
                }
              >
                Clouds
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 0,
                  "width": "20%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "textAlign": "left",
                  }
                }
              >
                62
                °
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "marginBottom": 5,
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 0,
                  "width": "40%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "textAlign": "left",
                  }
                }
              >
                Monday
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 0,
                  "width": "20%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "textAlign": "left",
                  }
                }
              >
                PM
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 0,
                  "width": "20%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "textAlign": "left",
                  }
                }
              >
                Clouds
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 0,
                  "width": "20%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "textAlign": "left",
                  }
                }
              >
                73
                °
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaView>
`;

exports[`renders loading state correctly 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "backgroundColor": "#b9e1f8",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    testID="App-ActivityIndicator"
  >
    <ActivityIndicator
      animating={true}
      color="#999999"
      hidesWhenStopped={true}
      size="large"
    />
  </View>
</RNCSafeAreaView>
`;
